cli/src/main/groovy/io/micronaut/cli/profile/commands/AbstractCreateCommand.groovy

```diff
@@ -477,6 +478,11 @@ abstract class AbstractCreateCommand extends ArgumentCompletingCommand implement
             tokens.put("mainClass", mainClass ? mainClass : "${this.defaultpackagename}.Application")
         }
 
+        // Allows for a separate custom main in native-image.properties
+        if (features.any{ it.name == "graal-native-image"} ) {
+            tokens.put("graalFunctionMainClass", "io.micronaut.function.executor.FunctionApplication")
+        }
```

Comment:
Keep in mind that this PR https://github.com/micronaut-projects/micronaut-profiles/pull/206 adds the feature `graal-native-image-kotlin` so that value should be checked here.

Reply:
Ok I added that option and it doesnt break anything but cant be fully tested till https://github.com/micronaut-projects/micronaut-profiles/pull/206 is merged